src/window.c

```diff
@@ -221,6 +227,18 @@ wset_update_mode_line (struct window *w)
     w->update_mode_line = true;
 }
 
+struct glyph_matrix*
+wget_current_matrix(const struct window *w)
+{
+  return w->current_matrix;
+}
+
+int
+wget_mode_line_height(const struct window *w)
+{
+  return w->mode_line_height;
+}
```

Comment:
why are you adding things written in C?

Reply:
To get around the bitfields. This is how we have been handling them. The bitfields in C structs are highly compiler dependent.